### PR TITLE
Fix a crash caused by invalid field intensity levels

### PR DESCRIPTION
**Pull request by Qrox**
#### Summary
SUMMARY: Bugfixes "Fix a crash caused by invalid field intensity levels"

#### Purpose of change
The mapgen code places smoke vent with level 2 intensity whose definition is missing from json, and the `field_type` getters did not check for the supplied intensity level parameter, resulting in crash if, for example, the tile containining the invalid field is examined.

#### Describe the solution
The supplied intensity level is now checked for validity, and fallback values are returned if the level is invalid.

#### Additional context
Also cleared up the debug field editor code a bit.

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
It should fallback to max defined intensity level.

**Comment by Qrox** [comment]
Yes, I also found that out. Anyway, this change is still good for those older saves which may contain incorrect field intensity levels (such as mine). I also made it fallback to max defined intensity level as you suggested.

**Comment by ZhilkinSerg** [comment]
Oh, I didn't think of existing saves. Yes, we need to add limit check for savegame loading.

**Comment by Qrox** [comment]
Restored smoke vent intensity as it now seems correct after investigation into #32935.

File: src/field_type.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct field_intensity_level {
     int extra_radiation_max = 0;
     float light_emitted = 0.0f;
     float translucency = 0.0f;
-    int convection_temperature_mod = 0.0f;
+    int convection_temperature_mod = 0;
```
Comment (ZhilkinSerg) [inline src/field_type.h]:
```suggestion
    int convection_temperature_mod = 0.0f;
```

File: src/field_type.cpp [hunk 2]
```diff
@@ -63,6 +63,20 @@ int_id<field_type>::int_id( const string_id<field_type> &id ) : _id( id.id() )
 {
 }
 
+const field_intensity_level &field_type::get_intensity_level( int level ) const
+{
+    if( intensity_levels.empty() ) {
```
Comment (ZhilkinSerg) [inline src/field_type.cpp]: To appease clang builds I am removing emptiness check clause in #33024 - it is already done during json loading anyway and it seems redundant to me here, but please tell me if you have other concerns about that.

**Pull request merged**